assets/app/scripts/directives/logViewer.js

```diff
@@ -233,16 +233,8 @@ angular.module('openshiftConsole')
             APIDiscovery
               .getLoggingURL()
               .then(function(url) {
-                // TODO: update lodash for _.get, would rather do this:
-                // var projectName = _.get($scope, 'context', 'project', 'metadata', 'name');
-                // var containerName = _.get($scope, 'options', 'container');
-                var projectName = $scope.context &&
-                                  $scope.context.project &&
-                                  $scope.context.project.metadata &&
-                                  $scope.context.project.metadata.name;
-
-                var containerName = $scope.options &&
-                                    $scope.options.container;
+                var projectName = _.get($scope.context, ['project', 'metadata', 'name']);
+                var containerName = _.get($scope.options, 'container');
```

Comment:
if that works too thats definitely easier to read

On Tuesday, January 26, 2016, Sam Padgett <notifications@github.com> wrote:

> In assets/app/scripts/directives/logViewer.js
> <https://github.com/openshift/origin/pull/6824#discussion_r50844857>:
>
> > @@ -233,16 +233,8 @@ angular.module('openshiftConsole')
> >              APIDiscovery
> >                .getLoggingURL()
> >                .then(function(url) {
> > -                // TODO: update lodash for _.get, would rather do this:
> > -                // var projectName = _.get($scope, 'context', 'project', 'metadata', 'name');
> > -                // var containerName = _.get($scope, 'options', 'container');
> > -                var projectName = $scope.context &&
> > -                                  $scope.context.project &&
> > -                                  $scope.context.project.metadata &&
> > -                                  $scope.context.project.metadata.name;
> > -
> > -                var containerName = $scope.options &&
> > -                                    $scope.options.container;
> > +                var projectName = _.get($scope.context, ['project', 'metadata', 'name']);
> > +                var containerName = _.get($scope.options, 'container');
>
> Just curious if there is any reason to use an array vs a string?
>
> var projectName = _.get($scope.context, 'project.metadata.name');
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6824/files#r50844857>.
>